### PR TITLE
[3.10] bpo-31746: Prevent segfaults when sqlite3.Connection is uninitialised (GH-27431)

### PR DESCRIPTION
**Pull request by erlend-aasland**
(cherry picked from commit 7e311e496b0e26b3d3c62fe9b0ed2a4677c37ee9)

Co-authored-by: Erlend Egeberg Aasland <erlend.aasland@innova.no>

https://bugs.python.org/issue31746

### CODE REVIEW

**Comment by erlend-aasland** [comment]
cc. @pablogsal

**Comment by erlend-aasland** [comment]
Thanks, @ambv 🙏🏻

File: Modules/_sqlite/connection.c [hunk 1]
```diff
@@ -1295,11 +1304,17 @@ pysqlite_connection_set_isolation_level(pysqlite_Connection* self, PyObject* iso
...
-            return -1;
+        /* We might get called during connection init, so we cannot use
+         * pysqlite_connection_commit() here. */
+        if (self->db && !sqlite3_get_autocommit(self->db)) {
+            int rc;
+            Py_BEGIN_ALLOW_THREADS
+            rc = sqlite3_exec(self->db, "COMMIT", NULL, NULL, NULL);
+            Py_END_ALLOW_THREADS
+            if (rc != SQLITE_OK) {
+                return _pysqlite_seterror(self->db, NULL);
```
Comment (ambv) [inline Modules/_sqlite/connection.c:1315]: 👍🏻

Confirmed that `_pysqlite_seterror` is indeed different in `3.10` compared to `main`:
https://github.com/python/cpython/blob/3.10/Modules/_sqlite/util.h#L38

Comment (erlend-aasland) [reply Modules/_sqlite/connection.c:1315]: Yes, it was changed in ~GH-26535~ GH-25915, which was not backported.

**Pull request merged**